src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Serialization
+{
+    internal sealed class SerializableSourceText
+    {
+        public ITemporaryTextStorageWithName? Storage { get; }
+        public SourceText? Text { get; }
+
+        public SerializableSourceText(ITemporaryTextStorageWithName storage)
+            : this(storage, text: null)
+        {
+        }
+
+        public SerializableSourceText(SourceText text)
+            : this(storage: null, text)
+        {
+        }
+
+        private SerializableSourceText(ITemporaryTextStorageWithName? storage, SourceText? text)
+        {
+            Debug.Assert(storage is not null || text is not null);
+
+            Storage = storage;
+            Text = text;
+        }
+
+        public bool IsDefault => Storage is null && Text is null;
```

Comment:
feels like this coudl never be true.  there are only two public entrypoints.  one requires Storage, one requires Text.  If that's not so, then something is super unclear to me and docs might help.

Reply:
I was thinking this was a value type for some reason.

Reply:
➡️ Fixed by removing this property